browser/ai_chat/sources.gni

```diff
@@ -4,6 +4,8 @@
 # You can obtain one at https://mozilla.org/MPL/2.0/.
 
 brave_browser_ai_chat_sources = [
+  "//brave/browser/ai_chat/ai_chat_service_factory.cc",
+  "//brave/browser/ai_chat/ai_chat_service_factory.h",
```

Comment:
I just tried this, but the problem is:
```
> gn check /Users/petemill/Development/Brave/brave-browser/src/out/Component_arm64
ERROR Dependency cycle:
  //chrome/browser:browser ->
  //brave/browser/ai_chat:ai_chat ->
  //chrome/browser:browser
```

And that dependency would be needed because:
- We haven't moved brave/browser/skus and brave/browser/misc_metrics to their own targets
- The browser needs the AIChatServiceFactory

And if I try to remove the dependency from brave/browser and only put it on brave/browser/ui I still get a gn_check error:
```
ERROR Dependency cycle:
  //chrome/browser:browser ->
  //brave/browser/brave_wallet:brave_wallet_delegate ->
  //chrome/browser/ui:ui ->
  //brave/browser/ui:ui ->
  //brave/browser/ai_chat:ai_chat ->
  //chrome/browser:browser
```

This seems like something that's worth untangling as a whole over the whole codebase and in a separate PR?

Reply:
Hm, yea, that's not ideal, it should only be //chrome/browser -> //brave/browser/ai_chat and not the other way around. That's an exact example why being in browser target itself would complicate things. :(

Reply:
I'm not seeing why AIChatServiceFactory would need to depend on `//chrome/browser`. browser_process.h, brave_browser_process.h, profile_keyed_service_factory.h and profile.h are all in other targets specifically to fix these kinds of problems.

Reply:
oic, you mentioned misc_metrics. We definitely should not be adding to the problem here or it's just going to be that much harder to untangle cc @DJAndries to fix the issue with misc_metrics. 
cc @bsclifton because SkusServiceFactory does not have any deps that would require it to be in sources.gni

Reply:
I guess this PR was opened before we added codeowner for sources.gni because we're using that to get this issue under control

Reply:
These both look like easy fixes, I don't see any reason why they needed to be in sources.gni in the first place. If there are any problems just split out interface/impl and put the header in its own target